public/js/reader.js

```diff
@@ -275,83 +324,81 @@ function initArchivePageOverlay() {
     $("#archivePagesOverlay").attr("loaded", "true");
 }
 
-function applySettings() {
+Reader.toggleFitMode = function (e) {
+    // possible options: fit-container, fit-width, fit-height
+    Reader.fitMode = localStorage.fitMode = e.target.id;
+    $("#fit-mode input").removeClass("toggled");
+    $(e.target).addClass("toggled");
 
-    $(".favtag-btn").removeClass("toggled");
-    $("#containersetting").hide();
-
-    if (!isNaN(localStorage.containerwidth))
-        $("#containerwidth").val(localStorage.containerwidth);
-
-    if (localStorage.readorder === "true")
-        $("#mangaread").addClass("toggled");
-    else
-        $("#normalread").addClass("toggled");
-
-    if (localStorage.doublepage === "true")
-        $("#doublepage").addClass("toggled");
-    else
-        $("#singlepage").addClass("toggled");
-
-    if (localStorage.forcefullwidth === "true")
-        $("#fitwidth").addClass("toggled");
-    else if (localStorage.scaletoview === "true")
-        $("#fitheight").addClass("toggled");
-    else {
-        $("#fitcontainer").addClass("toggled");
-        $("#containersetting").show();
+    if (Reader.fitMode === "fit-container") {
+        $("#container-width").show();
+    } else {
+        $("#container-width").hide();
     }
+    Reader.applyContainerWidth();
+};
 
-    if (localStorage.hidetop === "true")
-        $("#hidetop").addClass("toggled");
-    else
-        $("#showtop").addClass("toggled");
+Reader.registerContainerWidth = function () {
+    // Examples of allowed values: 1200, 1200px, 90%
+    // Default value: 1200px
+    const raw = $("#container-width-input").val().trim();
+    if (!raw) { // fall back to default
+        delete Reader.containerWidth;
+        localStorage.removeItem("containerWidth");
+    } else {
+        let value, type;
 
-    if (localStorage.nobookmark === "true")
-        $("#nobookmark").addClass("toggled");
-    else
-        $("#dobookmark").addClass("toggled");
+        [, value, type] = /^(\d+)(px|%)?$/.exec(raw);
+        value ||= 1200;
+        type ||= "px";
 
-    // Reset reader
-    goToPage(currentPage);
-}
+        Reader.containerWidth = localStorage.containerWidth = `${value}${type}`;
+    }
+    Reader.applyContainerWidth();
+};
 
-function setDisplayMode(fittowidth, fittoheight) {
-    localStorage.forcefullwidth = fittowidth;
-    localStorage.scaletoview = fittoheight;
-    applySettings();
-}
+Reader.applyContainerWidth = function () {
+    $("#img, .sni").attr("style", "");
 
-function setDoublePage(doublepage) {
-    localStorage.doublepage = doublepage;
-    applySettings();
-}
+    if (Reader.fitMode === "fit-height") {
```

Comment:
It'd be better imo to keep some of the comments from the previous code in this, some of the behavior could stand to stay documented. 

Even though it's way more explicit thanks to the better variable names, stuff like max-width jumping to 90% in double page or max-height being 98vh instead of 90 if the header is hidden is still a bit obscure to first-comers...

Reply:
Added comments documenting the fit behavior, as requested.